lib/src/main/java/edu/stanford/futuredata/macrobase/datamodel/Row.java

```diff
@@ -55,11 +55,16 @@ public int hashCode() {
     public String toString() {
         ArrayList<String> strs = new ArrayList<>(vals.size());
         for (Object o : vals) {
-            String s = o.toString();
-            if (o instanceof Double) {
-                if (s.length() > 7) {
-                    double v = ((Double) o).doubleValue();
-                    s = String.format("%.6g", v);
+            String s;
+            if (o == null) {
+                s = "NULL";
+            } else {
+                s = o.toString();
+                if (o instanceof Double) {
+                    if (s.length() > 7) {
```

Comment:
Just adjusted it until the floating point numbers were a reasonable length, 1.3e-10 is already 7 characters.